### PR TITLE
feat: introduce handler lock to edits (hl-1149)

### PR DESCRIPTION
**Pull request by sirtawast**
## Description :sparkles:

* Only one handler at a time
* Assign handler when status is changed to `handling` (clicked in GUI by handler)
* Disable edits by wrong handler with an overlay
* Any handler can change the assigned handler by clicking the `ApplicationHeader` unlock button
* In case GUI does not update because of simultaneous locks/unlocks, edits are prevented in backend level as well
* New API endpoint to do the unlocks because we can't send any old data using `POST /v1/handlerapplications/{uuid}/`

image

### CODE REVIEW

**Comment by terovirtanen** [comment]
**ADMIN branch is deployed to platta: https://tet-admin-ui-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tet-admin-ui-pr2945.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**YOUTH branch is deployed to platta: https://tet-youth-ui-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tet-youth-ui-pr2945.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://kesaseteli-pr2945.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://tet-pr2945.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**EMPLOYER branch is deployed to platta: https://kesaseteli-employer-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2945.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**ADMIN branch is deployed to platta: https://tet-admin-ui-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tet-admin-ui-pr2945.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr2945.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-employer-pr2945.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe handler result is __failed__ for https://helsinkilisa-ui-handler-pr2945.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://kesaseteli-handler-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-handler-pr2945.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**YOUTH branch is deployed to platta: https://tet-youth-ui-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tet-youth-ui-pr2945.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**YOUTH branch is deployed to platta: https://kesaseteli-youth-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-youth-pr2945.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://tet-pr2945.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2945.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**ADMIN branch is deployed to platta: https://tet-admin-ui-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**EMPLOYER branch is deployed to platta: https://kesaseteli-employer-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-handler-pr2945.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tet-admin-ui-pr2945.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr2945.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-employer-pr2945.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://kesaseteli-handler-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-handler-pr2945.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**YOUTH branch is deployed to platta: https://tet-youth-ui-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**YOUTH branch is deployed to platta: https://kesaseteli-youth-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-youth-pr2945.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://kesaseteli-pr2945.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://tet-pr2945.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**EMPLOYER branch is deployed to platta: https://kesaseteli-employer-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-employer-pr2945.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2945.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr2945.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-handler-pr2945.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://kesaseteli-handler-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-handler-pr2945.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**YOUTH branch is deployed to platta: https://kesaseteli-youth-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-youth-pr2945.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://kesaseteli-pr2945.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://tet-pr2945.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**EMPLOYER branch is deployed to platta: https://kesaseteli-employer-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**ADMIN branch is deployed to platta: https://tet-admin-ui-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tet-admin-ui-pr2945.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-employer-pr2945.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-handler-pr2945.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**YOUTH branch is deployed to platta: https://tet-youth-ui-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tet-youth-ui-pr2945.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**YOUTH branch is deployed to platta: https://kesaseteli-youth-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-youth-pr2945.dev.hel.ninja :laughing::tada::tada::tada:

File: frontend/benefit/handler/src/components/applicationReview/ApplicationReview.sc.ts [hunk 1]
```diff
@@ -15,10 +15,32 @@ type TabButtonProps = {
   active?: boolean;
 };
 
-export const $ApplicationReview = styled.div`
+type ApplicationReviewProps = {
```
Comment (EmiliaMakelaVincit) [inline frontend/benefit/handler/src/components/applicationReview/ApplicationReview.sc.ts:18]: I would only add extra props for styled components if they directly affected the resulting style. That's not the case currently and this `editDisabled` is not used at all; in `ApplicationReview` itself, all uses of `editDisabled` come directly from the constant received from `useApplicationReview`.

File: frontend/benefit/handler/src/components/applicationReview/ApplicationReview.sc.ts [hunk 2]
```diff
@@ -122,6 +144,12 @@ export const $Highlight = styled.div`
 `;
 
 export const $InfoNeededBar = styled.div`
+  position: relative;
+  display: flex;
+  flex-flow: row wrap;
+  justify-content: center;
+  align-items: center;
+  z-index: 999
```
Comment (EmiliaMakelaVincit) [inline frontend/benefit/handler/src/components/applicationReview/ApplicationReview.sc.ts:152]: Semicolon missing here, probably effectively disabling both this and the following CSS rule.

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://kesaseteli-pr2945.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2945.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**ADMIN branch is deployed to platta: https://tet-admin-ui-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-handler-pr2945.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tet-youth-ui-pr2945.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://kesaseteli-pr2945.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://tet-pr2945.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2945.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**ADMIN branch is deployed to platta: https://tet-admin-ui-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**EMPLOYER branch is deployed to platta: https://kesaseteli-employer-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tet-admin-ui-pr2945.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-handler-pr2945.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr2945.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-employer-pr2945.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://kesaseteli-handler-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-handler-pr2945.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**YOUTH branch is deployed to platta: https://tet-youth-ui-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tet-youth-ui-pr2945.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**YOUTH branch is deployed to platta: https://kesaseteli-youth-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

File: frontend/benefit/handler/src/components/applicationReview/ApplicationReview.sc.ts (hunk 1 above)
Comment (sirtawast) [reply frontend/benefit/handler/src/components/applicationReview/ApplicationReview.sc.ts:18]: Yeah, I had a &:before element there previously which used the prop on for `display`. Will remove this.

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-youth-pr2945.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://kesaseteli-pr2945.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://tet-pr2945.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**EMPLOYER branch is deployed to platta: https://kesaseteli-employer-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**ADMIN branch is deployed to platta: https://tet-admin-ui-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2945.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tet-admin-ui-pr2945.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-employer-pr2945.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr2945.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe handler result is __failed__ for https://helsinkilisa-ui-handler-pr2945.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-handler-pr2945.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://kesaseteli-handler-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-handler-pr2945.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**YOUTH branch is deployed to platta: https://tet-youth-ui-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://tet-youth-ui-pr2945.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**YOUTH branch is deployed to platta: https://kesaseteli-youth-pr2945.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-youth-pr2945.dev.hel.ninja :laughing::tada::tada::tada:

**Pull request merged**